src/parser/jobs/ast/modules/CrownPlay.tsx

```diff
@@ -0,0 +1,250 @@
+			this.minorArcanaDrift = event.timestamp - this.parser.pull.timestamp
+		} else {
+			// Take holding as from the time it comes off cooldown
+			this.minorArcanaDrift = event.timestamp - this.lastCrownTimestamp - this.data.actions.MINOR_ARCANA.cooldown
+		}
+
+		// Keep track of total drift time not using crown play
+		if (this.minorArcanaDrift > 0) {
+			this.minorArcanaTotalDrift += this.minorArcanaDrift
+		}
+
+		//catch any cards that were overwritten on the last cast. this only happens if AST redraws the same arcana
+		if (this.lastEventType === event.type) {
+			if (this.lastCardType === this.data.statuses.LORD_OF_CROWNS_DRAWN.id) { this.lordObtained++ }
+			if (this.lastCardType === this.data.statuses.LADY_OF_CROWNS_DRAWN.id) { this.ladyObtained++ }
+		}
+
+		// update the last use
+		this.lastCrownTimestamp = event.timestamp
+		this.lastEventType = event.type
+	}
+
+	private onCrownAction(event: Events['action']) {
+		if (event.action === this.data.actions.LORD_OF_CROWNS.id) {
+			this.lordCasts++
+			//to catch prepulls
+			if (this.lordObtained === 0 && this.parser.pull.timestamp + this.data.actions.MINOR_ARCANA.cooldown < event.timestamp) { this.lordObtained++ }
+		}
+		if (event.action === this.data.actions.LADY_OF_CROWNS.id) {
+			this.ladyCasts++
+			//to catch prepulls
+			if (this.ladyObtained === 0 && this.parser.pull.timestamp + this.data.actions.MINOR_ARCANA.cooldown < event.timestamp) { this.ladyObtained++ }
+		}
+	}
+
+	private onCrownBuff(event: Events['statusApply']) {
+		if (event.status === this.data.statuses.LORD_OF_CROWNS_DRAWN.id) {
+			this.lordObtained++
+			this.lastCardType = this.data.statuses.LORD_OF_CROWNS_DRAWN.id
+			this.lastEventType = event.type
+		}
+		if (event.status === this.data.statuses.LADY_OF_CROWNS_DRAWN.id) {
+			this.ladyObtained++
+			this.lastCardType = this.data.statuses.LADY_OF_CROWNS_DRAWN.id
+			this.lastEventType = event.type
+		}
+	}
+
+	private onComplete() {
+
+		// If they stopped using crown play at any point in the fight, this'll calculate the drift "accurately"
+		const fightEnd = this.parser.pull.duration + this.parser.pull.timestamp
+		if (this.lastCrownTimestamp === 0) {
+			this.minorArcanaTotalDrift = this.parser.pull.duration
+		} else {
+			this.minorArcanaTotalDrift += Math.max(0, fightEnd - this.lastCrownTimestamp - this.data.actions.MINOR_ARCANA.cooldown)
+		}
+
+		//if they overwrote a card but didn't play another, this will catch the last cast
+		if (this.lastEventType === 'action') {
+			if (this.lastCardType === this.data.statuses.LORD_OF_CROWNS_DRAWN.id) { this.lordObtained++ }
+			if (this.lastCardType === this.data.statuses.LADY_OF_CROWNS_DRAWN.id) { this.ladyObtained++ }
+		}
+
+		// Begin Theoretical Max Plays calc		//assumes that crown-play is not on cooldown at the start of the fight
+		const playsFromCrownPlay = Math.ceil(Math.max(0, (this.parser.pull.duration - oGCD_ALLOWANCE)) / this.data.actions.MINOR_ARCANA.cooldown)
```

Comment:
I think this formula still holds since it would assume everything went perfectly and it wouldn't be on CD due to last pull before boss or something
I can remove the comment if you prefer based on discussion above
Lemme know!